pytensor/tensor/rewriting/linalg.py

```diff
@@ -539,3 +540,64 @@ def svd_uv_merge(fgraph, node):
                     or len(fgraph.clients[cl.outputs[2]]) > 0
                 ):
                     return [cl.outputs[1]]
+
+
+def _find_solve_with_eye(node):
+    valid_solves = (Solve, SolveTriangular)
+    # First, we look for the solve op
+    if not (
+        isinstance(node.op, Blockwise) and isinstance(node.op.core_op, valid_solves)
+    ):
+        return False
+    # Check whether second input to solve is Eye
+    solve_inputs = node.inputs
+    eye_input = solve_inputs[1].owner
+
+    if not (eye_input and isinstance(eye_input.op, Eye)):
+        return False
+
+    if eye_input.inputs[-1].data.item() != 0:
+        return False
+    return True
+
+
+@register_canonicalize
+@register_stabilize
+@node_rewriter([Blockwise])
+def rewrite_inv_inv(fgraph, node):
+    valid_inverses = (MatrixInverse, MatrixPinv, Solve, SolveTriangular)
+    valid_solves = (Solve, SolveTriangular)
+    # Check if its a valid inverse operation (either inv/pinv or if its solve, then b = eye)
+    inv_check = False
+    if isinstance(node.op, Blockwise) and isinstance(node.op.core_op, valid_inverses):
+        inv_check = True
+        if isinstance(node.op.core_op, valid_solves):
+            inv_check = _find_solve_with_eye(node)
+
+    if not inv_check:
+        return None
+
+    potential_inner_inv = node.inputs[0].owner
+    if potential_inner_inv is None or potential_inner_inv.op is None:
+        return None
+
+    # Check if its a valid inverse operation (either inv/pinv or if its solve, then b = eye)
```

Comment:
This comment is just a repeat of the previous one? I'd like to see all of these if checks clearly commented, it's not at all obvious how or why they work